src/azure-cli/azure/cli/command_modules/acs/_params.py

```diff
@@ -551,6 +551,7 @@ def load_arguments(self, _):
         c.argument('node_osdisk_type', arg_type=get_enum_type(node_os_disk_types))
         c.argument('node_osdisk_size', type=int)
         c.argument('max_surge', validator=validate_max_surge)
+        c.argument('drain_timeout', type=int) #not bothering to validate posiive since RP will handle that though zero will just be omitted.
```

Comment:
need to add a validator if you want to verify that the value is positive

Reply:
please add a new live test case (see test_aks_commands.py) for this option

Reply:
I don't thanks!

Reply:
Removing this comment. Prefer these validations to happen at server side so if we cahnge them we also don't need to release no cli. Since they're on frontend they're pretty fast





az aks nodepool update  -g paultest --cluster-name draintimeout -n nodepool1 --drain-timeout -5

(InvalidParameter) The value of parameter agentPoolProfile.upgrade.drainTimeout is invalid. Error details: Drain timeout must be between 1 and 1440 minutes. Please see https://aka.ms/aks-naming-rules for more details.
Code: InvalidParameter